### PR TITLE
 Add support for b'a', b"a" and c"a"

### LINKED ISSUE #106: Add support for `b'a'`, `b"a"` and `c"a"`
It would allow things like: `if let [b'H', b'i', b' ', who @ .., b'!'] = b"Hi world!"`.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #106.

### CODE REVIEW

File: rinja_parser/src/lib.rs [hunk 1]
```diff
@@ -425,15 +459,35 @@ fn str_lit(i: &str) -> ParseResult<'_> {
...
+        Some('b') => StrPrefix::Binary,
+        Some('c') => StrPrefix::CLike,
+        _ => StrPrefix::None,
+    };
+    Ok((i, StrLit { prefix, content }))
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CharLit<'a> {
+    pub has_b_prefix: bool,
```
Comment (Kijewski) [inline rinja_parser/src/lib.rs:475]: Could you please make this a `prefix: CharPrefix` field analogous to `StrLit`? I am hopeful that sometime soon ASCII literals like `a"Hello"` or `a'!'` are introduced.

File: rinja_parser/src/lib.rs [hunk 2]
```diff
@@ -416,7 +416,41 @@ fn separated_digits(radix: u32, start: bool) -> impl Fn(&str) -> ParseResult<'_>
     }
 }
 
-fn str_lit(i: &str) -> ParseResult<'_> {
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum StrPrefix {
+    None,
```
Comment (Kijewski) [inline rinja_parser/src/lib.rs:421]: I think the code would be more readable `StrLit` used `Option<StrPrefix>`, and this variant was removed.

Comment (GuillaumeGomez) [reply rinja_parser/src/lib.rs:421]: I originally did that. Didn't like it for some reason but I think it's more logical to do as you suggest. :+1:

File: rinja_parser/src/lib.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply rinja_parser/src/lib.rs:475]: Coherency and also easier for future updates. Love it!

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by Kijewski — APPROVED** [review]
Looks good, thank you!

**Pull request merged**